src/SharedUtilities/Domains.jl

```diff
@@ -16,14 +16,17 @@ An abstract type for domains.
 """
 abstract type AbstractDomain{FT <: AbstractFloat} end
 Base.eltype(::AbstractDomain{FT}) where {FT} = FT
+
 """
     coordinates(domain::AbstractDomain)
 
-Method which returns the coordinates appropriate for a given domain.
-
-The coordinates can be Fields or Vectors.
+Returns the coordinate field for the domain.
 """
-function coordinates(domain::AbstractDomain) end
+function coordinates(domain::AbstractDomain)
```

Comment:
As we discussed, `coordinates` still calls `make_function_space`. It should be called only once during the initialization step.

Reply:
sounds good!